improver_tests/psychrometric_calculations/hail_size/test_hail_size.py

```diff
@@ -312,3 +280,80 @@ def test_model_id_attr(
         relative_humidity_on_pressure,
     )
     metadata_check(result)
+    cube_size_check(result)
+
+
+def test_re_ordered_cubes(
+    temperature_on_pressure_levels,
+    ccl_pressure,
+    relative_humidity_on_pressure,
+    ccl_temperature,
+):
+
+    """Tests Plugin is input cubes have coordinates in a different order.
+    Checks that the outputted cube has coordinates in the same order as the inputs"""
+
+    enforce_coordinate_ordering(
+        temperature_on_pressure_levels,
+        ["pressure", "latitude", "realization", "longitude"],
+    )
+    enforce_coordinate_ordering(
+        relative_humidity_on_pressure,
+        ["pressure", "latitude", "realization", "longitude"],
+    )
+    enforce_coordinate_ordering(ccl_pressure, ["latitude", "realization", "longitude"])
+    enforce_coordinate_ordering(
+        ccl_temperature, ["latitude", "realization", "longitude"]
+    )
+    result = HailSize()(
+        ccl_temperature,
+        ccl_pressure,
+        temperature_on_pressure_levels,
+        relative_humidity_on_pressure,
+    )
+
+    metadata_check(result)
+    coord_names = [coord.name() for coord in result.coords()]
+    assert coord_names == [
+        "latitude",
+        "realization",
+        "longitude",
+        "forecast_period",
+        "forecast_reference_time",
+        "time",
+    ]
+    assert result.shape == (3, 2, 2)
+
+
+def test_no_realization_coordinate(
+    temperature_on_pressure_levels,
+    ccl_pressure,
+    relative_humidity_on_pressure,
+    ccl_temperature,
+):
+    """Test plugin if input cubes don't have a realization coordinate"""
+
+    temp = next(temperature_on_pressure_levels.slices_over("realization"))
+    temp.remove_coord("realization")
+
+    humidity = next(relative_humidity_on_pressure.slices_over("realization"))
+    humidity.remove_coord("realization")
+
+    cloud_pressure = next(ccl_pressure.slices_over("realization"))
+    cloud_pressure.remove_coord("realization")
+
+    cloud_temp = next(ccl_temperature.slices_over("realization"))
+    cloud_temp.remove_coord("realization")
+
+    result = HailSize()(cloud_temp, cloud_pressure, temp, humidity)
+
+    metadata_check(result)
+    coord_names = [coord.name() for coord in result.coords()]
+    assert coord_names == [
+        "latitude",
+        "longitude",
+        "forecast_period",
+        "forecast_reference_time",
+        "time",
+    ]
+    assert result.shape == (3, 2)
```

Comment:
Is it worth testing the results are correct in this case as well?

Reply:
Good thought (actually showed a small problem with the code so definately worth doing).